agent/structs/config_entry.go

```diff
@@ -479,7 +479,12 @@ func (c *ConfigEntryResponse) MarshalBinary() (data []byte, err error) {
 	bs := make([]byte, 128)
 	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
 
-	if err := enc.Encode(c.Entry.GetKind()); err != nil {
+	kind := ""
+	if c.Entry != nil {
+		kind = c.Entry.GetKind()
+	}
+
+	if err := enc.Encode(kind); err != nil {
 		return nil, err
 	}
 	if err := enc.Encode(c.Entry); err != nil {
```

Comment:
is it desirable for the nil to be encoded here, or should it be skipped?

Reply:
I guess we could elide it.